src/SignalR/server/SignalR/src/HubEndpointRouteBuilderExtensions.cs

```diff
@@ -46,6 +46,16 @@ public static class HubEndpointRouteBuilderExtensions
                                                 "'IServiceCollection.AddSignalR' inside the call to 'ConfigureServices(...)' in the application startup code.");
         }
 
+        using var scope1 = endpoints.ServiceProvider.CreateScope();
```

Comment:
Should this check only in Development Environment?

Reply:
I could see singleton hubs being okay strictly in test scenarios where the author knows there will be no concurrent connections or invocations. I imagine that's the main case where people would do this since it'd be easier to assert things about what happened with a singleton hub instance. Unfortunately, I'm not sure whether most people use the "Development" environment in their tests. I suspect many may not.